### PR TITLE
add a new section to introduce client-side video upscaling feature

### PR DESCRIPTION
**Pull request by adglkh**

### CODE REVIEW

**Review by jat-canonical — APPROVED** [review]
LGTM

**Review by ajanon — CHANGES_REQUESTED** [review]
One typo

**Review by ajanon — APPROVED** [review]
LGTM

**Review by keirthana — CHANGES_REQUESTED** [review]

File: howto/stream/enable-client-side-video-upscaling.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+(howto-enable-client-side-video-upscaling)=
+# Client side video upscaling
+
+The [Anbox Streaming SDK](https://github.com/canonical/anbox-streaming-sdk) offers experimental video upscaling features for client side streaming, leveraging [AMD FidelityFX Super Resolution 1.0](https://gpuopen.com/fidelityfx-superresolution/). This feature allows video to be transmitted at lower resolutions, saving bandwidth without losing too much detail. The SDK sets up a [WebGL](https://developer.mozilla.org/en-US/docs/Web/API/WebGL_API/Tutorial/Getting_started_with_WebGL) context and utilizes a fragment shader based on the FSR 1.0 algorithm for post-processing, producing clearer and sharper video frames before they are displayed on the screen.
+
+## How to enable client-side video upscaling
+
+To enable client-side video upscaling, initiating the `AnboxStream` object as follows:
```
[the remark this answers is not in the document]
Comment (adglkh) [reply howto/stream/enable-client-side-video-upscaling.md:8]: Will fix

**Comment by adglkh** [comment]
> Few minor corrections and one question:
> 
> For each of the actions, I feel like it would be more helpful if we mention _where_ to do the actions - for example, where would the user provide ``experimental.upscaling.useTargetFrameRate` option?


A: That may replicate what I put in the section `## How to enable client-side video upscaling` if we mention the usage for each option in action 
```html
<!DOCTYPE html>
    stream = new AnboxStream({
        targetElement: "anbox-stream",
        experimental: {
            upscaling: {
              enabled: true,
            },
        },
        ...
    });
```
However I don't have a strong opinion on this. I think that's better for a develop to get on this. I am fine with this too :)

File: howto/stream/enable-client-side-video-upscaling.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+(howto-enable-client-side-video-upscaling)=
+# Client side video upscaling
```
Comment (keirthana) [inline howto/stream/enable-client-side-video-upscaling.md:2]:
```suggestion
# How to enable clien-side video upscaling
```

File: howto/stream/enable-client-side-video-upscaling.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+The [Anbox Streaming SDK](https://github.com/canonical/anbox-streaming-sdk) offers experimental video upscaling features for client side streaming, leveraging [AMD FidelityFX Super Resolution 1.0](https://gpuopen.com/fidelityfx-superresolution/). This feature allows video to be transmitted at lower resolutions, saving bandwidth without losing too much detail. The SDK sets up a [WebGL](https://developer.mozilla.org/en-US/docs/Web/API/WebGL_API/Tutorial/Getting_started_with_WebGL) context and utilises a fragment shader based on the FSR 1.0 algorithm for post-processing, producing clearer and sharper video frames before they are displayed on the screen.
+
+## How to enable client-side video upscaling
```
Comment (keirthana) [inline howto/stream/enable-client-side-video-upscaling.md:6]:
```suggestion
## Initiate the `AnboxStream` object
```

File: howto/stream/enable-client-side-video-upscaling.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        experimental: {
+            upscaling: {
+              enabled: true,
+            },
+        },
+        ...
+    });
+```
+
+## Use target frame rate
```
Comment (keirthana) [inline howto/stream/enable-client-side-video-upscaling.md:23]:
```suggestion
### Use target frame rate
```

File: howto/stream/enable-client-side-video-upscaling.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+The option `experimental.upscaling.useTargetFrameRate` allows canvas to use the refresh frame rate setting in the Anbox container for rendering video frames. This avoids relying on `HTMLVideoElement.requestVideoFrameCallback` function, which can occasionally be fired [one v-sync late](https://wicg.github.io/video-rvfc/#introduction). This option is off by default and requires explicit activation.
+
+```{note}
+On a WebView where the `HTMLVideoElement.requestVideoFrameCallback` function is not supported, this option will be enabled by default.
+```
+
+## Use custom fragment shader
```
Comment (keirthana) [inline howto/stream/enable-client-side-video-upscaling.md:31]:
```suggestion
### Use custom fragment shader
```

**Comment by keirthana** [comment]
> > Few minor corrections and one question:
> > For each of the actions, I feel like it would be more helpful if we mention _where_ to do the actions - for example, where would the user provide ``experimental.upscaling.useTargetFrameRate` option?
> 
> A: That may replicate what I put in the section `## How to enable client-side video upscaling` if we mention the usage for each option in action

In that case, I have suggested some rearrangement in the form of heading levels which would make the flow a bit easier. Let me know if that is fine with you. 

> 
> ```
> <!DOCTYPE html>
>     stream = new AnboxStream({
>         targetElement: "anbox-stream",
>         experimental: {
>             upscaling: {
>               enabled: true,
>             },
>         },
>         ...
>     });
> ```
> 
> However I don't have a strong opinion on this. I think that's better for a develop to get on this. I am fine with this too :)

**Review by keirthana — CHANGES_REQUESTED** [review]

**Comment by adglkh** [comment]
> > > Few minor corrections and one question:
> > > For each of the actions, I feel like it would be more helpful if we mention _where_ to do the actions - for example, where would the user provide ``experimental.upscaling.useTargetFrameRate` option?
> > 
> > 
> > A: That may replicate what I put in the section `## How to enable client-side video upscaling` if we mention the usage for each option in action
> 
> In that case, I have suggested some rearrangement in the form of heading levels which would make the flow a bit easier. Let me know if that is fine with you.
A: that looks good to me as well.
> 
> > ```
> > <!DOCTYPE html>
> >     stream = new AnboxStream({
> >         targetElement: "anbox-stream",
> >         experimental: {
> >             upscaling: {
> >               enabled: true,
> >             },
> >         },
> >         ...
> >     });
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > However I don't have a strong opinion on this. I think that's better for a develop to get on this. I am fine with this too :)

File: howto/stream/enable-client-side-video-upscaling.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
+(howto-enable-client-side-video-upscaling)=
+# How to enable clien-side video upscaling
```
Comment (keirthana) [inline howto/stream/enable-client-side-video-upscaling.md:2]:
```suggestion
# How to enable client-side video upscaling
```

**Review by keirthana — APPROVED** [review]
One minor type, otherwise all looks good, thanks!

**Comment by keirthana** [comment]
@adglkh Added some words to ignore list for the spellcheck to pass, hope that is fine :crossed_fingers:

**Comment by adglkh** [comment]
@keirthana 
There are a few links broken, and we should be good now.
@morphis 
Could you please take a look?

Thanks

**Comment by keirthana** [comment]
> @keirthana There are a few links broken, and we should be good now. @morphis Could you please take a look?
> 

I added a [fix](https://github.com/canonical/anbox-cloud-docs/pull/103) for ignoring anchors (the link check was very flaky because of that), so should be good now.
> Thanks

**Review by morphis — APPROVED** [review]
LGTM

**Pull request merged**